opentracing/scope_managers/contextvars.py

```diff
@@ -0,0 +1,115 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from contextlib import contextmanager
+from contextvars import ContextVar
+
+from opentracing import Scope
+from opentracing.scope_managers import ThreadLocalScopeManager
+
+
+_SCOPE = ContextVar('scope')
+
+
+class ContextVarsScopeManager(ThreadLocalScopeManager):
+    """
+    :class:`~opentracing.ScopeManager` implementation for **asyncio**
+    that stores the :class:`~opentracing.Scope` using ContextVar.
+
+    The scope manager provides automatic :class:`~opentracing.Span` propagation
+    from parent coroutines to their children.
+
+    .. code-block:: python
+
+        async def child_coroutine(span):
+            # No need manual activation of parent span in child coroutine.
+            with tracer.start_active_span('child') as scope:
+                ...
+
+        async def parent_coroutine():
+            with tracer.start_active_span('parent') as scope:
+                ...
+                await child_coroutine(span)
+                ...
+
+    """
+    # TODO: update description
+
+    def activate(self, span, finish_on_close):
+        """
+        Make a :class:`~opentracing.Span` instance active.
+
+        :param span: the :class:`~opentracing.Span` that should become active.
+        :param finish_on_close: whether *span* should automatically be
+            finished when :meth:`Scope.close()` is called.
+
+        :return: a :class:`~opentracing.Scope` instance to control the end
+            of the active period for the :class:`~opentracing.Span`.
+            It is a programming error to neglect to call :meth:`Scope.close()`
+            on the returned instance.
+        """
+
+        return self._set_scope(span, finish_on_close)
+
+    @property
+    def active(self):
+        """
+        Return the currently active :class:`~opentracing.Scope` which
+        can be used to access the currently active
+        :attr:`Scope.span`.
+
+        :return: the :class:`~opentracing.Scope` that is active,
+            or ``None`` if not available.
+        """
+        # TODO: update description
+
+        return self._get_scope()
+
+    def _set_scope(self, span, finish_on_close):
+        return _ContextVarsScope(self, span, finish_on_close)
+
+    def _get_scope(self):
+        return _SCOPE.get(None)
+
+
+class _ContextVarsScope(Scope):
+    def __init__(self, manager, span, finish_on_close):
+        super(_ContextVarsScope, self).__init__(manager, span)
+        self._finish_on_close = finish_on_close
+        self._token = _SCOPE.set(self)
+
+    def close(self):
+        if self.manager.active is not self:
+            return
+        _SCOPE.reset(self._token)
+
+        if self._finish_on_close:
+            self.span.finish()
+
+
+@contextmanager
+def no_parent_scope():
```

Comment:
Any reason to not simply provide `default=None` for `_SCOPE`?

Reply:
I think it's not enough. I've added docstring to make the idea clear. Sometimes we can have a scope, that we don't want to propagate.